src/angular-cesium-widgets/services/zoom-to-rectangle.service.ts

```diff
@@ -0,0 +1,243 @@
+import { Injectable, Optional } from '@angular/core';
+import { MapsManagerService } from '../../angular-cesium/services/maps-manager/maps-manager.service';
+import { CameraService } from '../../angular-cesium/services/camera/camera.service';
+import { CesiumService } from '../../angular-cesium/services/cesium/cesium.service';
+import { AcMapComponent } from '../../angular-cesium/components/ac-map/ac-map.component';
+
+interface ZoomData {
+  container: HTMLElement;
+  borderElement?: HTMLElement;
+  resetOnEscapePressFunc?: EventListenerOrEventListenerObject;
+}
+
+/**
+ * The Service is as a "zoom to rectangle" tool
+ *
+ * example:
+ * ```
+ * constructor(
+ *   private cameraService: CameraService,
+ *   private cesiumService: CesiumService,
+ *   private zoomToRectangleService: ZoomToRectangleService,
+ * ) {
+ *   this.zoomToRectangleService.init(cesiumService, cameraService);
+ * }
+ * ...
+ * this.zoomToRectangleService.activate({onComplete: () => this.zoomToRectangleService.disable()});
+ * ```
+ *
+ * @function init - initialize the service with CameraService and CesiumService.
+ * If no mapId is provided to activate() - must be called before calling `activate()`.
+ *
+ * @function disable - disables the tool.
+ *
+ * @function activate -
+ * @param options
+ * {
+ *  onStart - optional - a callback that will be called when the user start drawing the rectangle
+ *  onComplete - optional - a callback that will be called when the tool zoom in
+ *  autoDisableOnZoom - optional - determines if the tool should auto disable after zoom - default: true
+ *  animationDurationInSeconds - optional - zoom animation duration in seconds - default: 0.5
+ *  borderStyle - optional - the style of the rectangle element border - default: '3px dashed #FFFFFF'
+ *  backgroundColor - optional - the background color of the rectangle element - default: 'transparent'
+ *  resetKeyCode - optional - the key code of the key that is used to reset the drawing of the rectangle - default: 27 (ESC key)
+ * }
+ * @param mapId - optional - the mapId of the map that the tool will be used in.
+ *
+ */
+
+@Injectable()
+export class ZoomToRectangleService {
+  private cameraService: CameraService;
+  private cesiumService: CesiumService;
+  constructor(
+    private mapsManager: MapsManagerService,
+    @Optional() cameraService: CameraService,
+    @Optional() cesiumService: CesiumService,
+  ) {}
+
+  init(cesiumService: CesiumService, cameraService: CameraService) {
+    this.cameraService = cameraService;
+    this.cesiumService = cesiumService;
+  }
+
+  private mapsZoomElements = new Map<string, ZoomData>();
+  private defaultOptions = {
+    animationDurationInSeconds: 0.5,
+    resetKeyCode: 27,
+    borderStyle: '2px solid rgba(0,0,0,0.5)',
+    backgroundColor: 'rgba(0,0,0,0.2)',
+    autoDisableOnZoom: true,
+  };
+
+  activate(
+    options: {
+      onStart?: (acMap?: AcMapComponent) => any;
```

Comment:
why the acMap is passed here?

Reply:
Because the service can be used without being injected under ac-map